proto/design/element/variable.proto

```diff
@@ -0,0 +1,94 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package com.android.designcompose.proto.design.element;
+
+import "design/element/color.proto";
+
+option java_multiple_files = true;
+
+// Representation of a Figma variable. We convert a figma_schema::Variable into
+// this format to make the fields a bit easier to access.
+message Variable {
+  // Variable Types
+  enum VariableType {
+    VARIABLE_TYPE_UNSPECIFIED = 0;
+    VARIABLE_TYPE_BOOL = 1;
+    VARIABLE_TYPE_NUMBER = 2;
+    VARIABLE_TYPE_TEXT = 3;
+    VARIABLE_TYPE_COLOR = 4;
+  }
+
+  string id = 1;
+  string name = 2;
+  bool remote = 3;
+  string key = 4;
+  string variable_collection_id = 5;
+  VariableType var_type = 6;
+  VariableValueMap values_by_mode = 7;
+}
+
+// We redeclare VariableValue instead of using the one from figma_schema because
+// the "untagged" attribute there prevents serde_reflection from being able to
+// run properly.
+message VariableValue {
+  oneof value {
+    bool boolean = 1;
+    float float = 2;
+    string string = 3;
+    Color color = 4;
+    VariableAlias alias = 5;
+  }
+}
+
+// Each variable contains a map of possible values. This data structure helps
+// keep track of that data and contains functions to retrieve the value of a
+// variable given a mode.
+message VariableValueMap {
+  map<string, VariableValue> values_by_mode = 1;
+}
+
+// a reference to a design variable (e.g., color, spacing). Think of it as a
+// symbolic name for a value that can change throughout your design.
+message VariableAlias {
+  string id = 1;
+  string name = 2;
```

Comment:
I don't think a VariableAlias has a name in it. I think it's just the ID, and we have to find the name based off the ID. Though I'm not sure how you're using this, maybe you've already resolved the name.

Reply:
How was this resolved?

Reply:
`VariableAlias` is structured the way it is because that is how it is parsed from Figma. The 'type' corresponds to the string "VARIABLE_ALIAS", such as:
```
            "boundVariables": {
              "itemSpacing": {
                "type": "VARIABLE_ALIAS",
                "id": "VariableID:2039:72"
              },
            }
```

Reply:
Wow, I'm sorry, Rob, I was not reading properly. I don't know why I looked at the VariableAlias message and thought that it had the same fields as the rust struct.

Ok, so you're saying that all we actually need is the ID. That makes sense. I also checked and BoundVariables doesn't get serialized (it's not in reflection.rs) so I shouldn't have a proto message for it anyways. I'll just remove the VariableAlias type since it's nothing more than a choice of the VariableValue enum/oneof.